File: Data Structures & Algorithms/combination-target-sum/submission-1.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        response = []

        def dfs(arr, i):
            if i >= len(nums):
                return
            if sum(arr) > target:
                return
            elif sum(arr) < target:
                arr.append(nums[i])
                dfs(arr, i)
                arr.pop()
                dfs(arr, i+1)
            else:
                response.append(arr.copy())
                return
        
        dfs([], 0)
        return response

File: Data Structures & Algorithms/maximum-subarray/submission-1.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int currMax = nums[0];
        int max = nums[0];

        for (int i{1uz}; i < nums.size(); ++i) {
            currMax = std::max(currMax + nums[i], nums[i]);
            max = std::max(currMax, max);
        }

        return max;
    }
};
